romancal/multiband_catalog/multiband_catalog_step.py

```diff
@@ -183,12 +198,117 @@ def process(self, library):
+            )
+
+            si_cen = SkyCoord(
+                ra=det_model.meta.wcsinfo.ra_ref * u.deg,
+                dec=det_model.meta.wcsinfo.dec_ref * u.deg,
+            )
+
+            # Obtain exposure times and filters
+            # This code assumes all filters have been coadded already,
+            # and thus there is one image per filter
+            si_filters = []
+            si_exptimes = {}
+            with library:
+                for model in library:
+                    si_filter_name = model.meta.instrument.optical_element
+
+                    si_exptimes[si_filter_name] = float(
+                        model.meta.coadd_info.exposure_time
+                    )
+                    si_filters.append(si_filter_name)
+
+                    library.shelve(model, modify=False)
+
+            # WCS object for ra & dec conversion
+            wcsobj = mk_gwcs(
+                det_model.meta.wcsinfo.ra_ref,
+                det_model.meta.wcsinfo.dec_ref,
+                det_model.meta.wcsinfo.roll_ref,
+                bounding_box=(
+                    (-0.5, model.data.shape[0] - 0.5),
+                    (-0.5, model.data.shape[1] - 0.5),
+                ),
+                shape=model.data.shape,
+            )
+
+            si_ra, si_dec = wcsobj.pixel_to_world_values(
+                np.array(si_x_pos), np.array(si_y_pos)
+            )
+
+            # Generate cosmos-like catalog
+            si_cat = injection.make_cosmoslike_catalog(
+                cen=si_cen,
+                ra=si_ra,
+                dec=si_dec,
+                exptimes=si_exptimes,
+            )
+
+            # Save catalog to segmentation image
+            segment_img.injected_sources = si_cat.as_array()
+
+            # Inject sources into the detection image
+            si_model = injection.inject_sources(si_model, si_cat)
+
+            # Create SI segmentation image
+            si_segment_img = make_segmentation_image(
+                si_model.data,
+                snr_threshold=self.snr_threshold,
+                npixels=self.npixels,
+                bkg_rms=si_bkg_rms,
+                deblend=self.deblend,
+                mask=mask,
+            )
+
+            # Save SI segmentation image
+            si_segment_img.detection_image = si_model.data
+            segment_img.si_detection_image = si_model.data
+
+            log.info("Creating catalog for source injected detection image")
+            si_det_catobj = RomanSourceCatalog(
+                si_model,
+                si_segment_img,
+                si_model.data,
+                star_kernel_fwhm,
+                fit_psf=self.fit_psf,
+                detection_cat=None,
+                mask=mask,
+                cat_type="dr_det",
+                ee_spline=ee_spline,
+            )
+
+            si_det_cat = si_det_catobj.catalog
+            si_det_cat.meta["ee_fractions"] = {}
+
+            segment_img.si_segment_img = si_segment_img
```

Comment:
This block has two conceptually separate bits:
- the source injection bits
- the "normal" preexisting processing bits.
  - background computation, segmentation image, source catalog construction, ee fractions
Ideally I think we'd factor out the normal processing bits and have this look more like:
```
make_catalog(original_image)
if inject_sources:
    injected_information = add_sources(...)
    make_catalog(injected_information)
```
I realize I'm oversimplifying, though!